backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerClient.java

```diff
@@ -53,23 +52,18 @@ public AnsibleRunnerClient() {
         this.returnValue = new AnsibleReturnValue(AnsibleReturnCode.ERROR);
     }
 
-    public Boolean playHasEnded(UUID uuid) {
-        String jobEvents = getJobEventsDir(uuid.toString());
-        File lastEventFile = new File(jobEvents + lastEvent);
-        String res = "";
-        try {
-                res = Files.readString(lastEventFile.toPath());
-            } catch (IOException e) {
-                return false;
-            }
-        return res.contains("playbook_on_stats");
+    public Boolean playHasEnded(String uuid, int lastEventId) {
+        // get current status, but new events might have appeared since last processing
+        PlaybookStatus currentPlaybookStatus = getPlaybookStatus(uuid);
+        String msg = currentPlaybookStatus.getMsg();
+        return !msg.equalsIgnoreCase("running") && lastEventId <= getTotalEvents(uuid);
```

Comment:
Shouldn't this be `lastEventId == getTotalEvents(uuid)` in the case when the playbook finish and the msg is no longer running we still process all events until the last one?

Reply:
oh sure, the other way around:)
&& !(lastEventId < getTotalEvents(uuid))

Reply:
uhhh the negative does not look best... if you don't like the `==` how about the `>=` ?